galley/pkg/config/analysis/msg/messages.yaml

```diff
@@ -504,3 +504,15 @@ messages:
         type: string
       - name: hosts
         type: string
+
+  - name: "ImageAutoWithoutInjection"
+    code: IST0146
+    level: Warning
```

Comment:
made warning since maybe be false positive if using `istioctl kube-inject` on the resources

Reply:
If we change to Pod, we can make it error

Reply:
Is it possible to warn on deployment, error on pod?  Deployment detection is useful for CI/CD use cases, where Pods will not yet exist.  But I would hate to see this ignored when pods have image:auto, which I have encountered before...

Reply:
Makes sense, changed it so that it warns on deployment and errors on pod, thanks!